src/sys/mman.rs

```diff
@@ -258,6 +258,27 @@ pub unsafe fn madvise(addr: *mut c_void, length: size_t, advise: MmapAdvise) ->
     Errno::result(libc::madvise(addr, length, advise as i32)).map(drop)
 }
 
+/// Calls to `mprotect` are inherently unsafe, as changes to memory protections can lead to
```

Comment:
The doc comment should start with a one-line description of the function.  This can optionallly be followed by a multiline description.  Safety information should be in a `#Safety` section.  There should also be a link to an online man page.  For standardized functions like this one, you should use the open group's man page: http://pubs.opengroup.org/onlinepubs/9699919799/functions/mprotect.html .